clang/test/SemaSYCL/intel-fpga-no-global-work-offset.cpp

```diff
@@ -61,7 +64,26 @@ int main() {
 
     h.single_task<class test_kernel8>(
         []() [[intel::no_global_work_offset(0),      // expected-note {{previous attribute is here}}
-	       intel::no_global_work_offset(1)]]{});  // expected-warning{{attribute 'no_global_work_offset' is already applied with different arguments}}
+	intel::no_global_work_offset(1)]]{});  // expected-warning{{attribute 'no_global_work_offset' is already applied with different arguments}}
+
+#if defined(SYCL2020)
+    // Test attribute is not propagated.
+    // CHECK-LABEL: FunctionDecl {{.*}}test_kernel9
+    // CHECK-NOT:   SYCLIntelNoGlobalWorkOffsetAttr {{.*}}
```

Comment:
Nothing is calling `FileCheck` in the RUN lines, so this isn't actually being tested.

Note: I'd like to see the AST node tests start being separated from the diagnostic tests. There's an `AST` directory where those should typically live, and it means we don't have to work around diagnostic tests that also check errors.

Reply:
The same issue seems to apply in other tests as well.

Reply:
i have updated Sema tests so that FileCheck can be tested in the RUN lines with both SYCL 2020 and SYCl 2017 modes.